kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaChanges.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.util;
+
+import io.delta.kernel.types.StructField;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * SchemaChanges encapsulates a list of added, removed, renamed, or updated fields in a schema
+ * change
+ */
+class SchemaChanges {
```

Comment:
I intentionally left this as package private. If we do want to include details on why validation failed and output the diff on failure to aid in debugging, I think we can handle that separately and internal to the validation path but I do think this class shouldn't be visible to folks.